tests/backup/backup_test.go

```diff
@@ -1937,35 +1968,63 @@ func getProviderClusterConfigPath(provider string, kubeconfigs []string) (string
 
 // CreateBackup creates backup
 func CreateBackup(backupName string, clusterName string, bLocation string, bLocationUID string,
-	namespaces []string, labelSelectors map[string]string, orgID string) {
-
-	Step(fmt.Sprintf("Create backup [%s] in org [%s] from cluster [%s]",
-		backupName, orgID, clusterName), func() {
+	namespaces []string, labelSelectors map[string]string, orgID string, uid string, pre_rule_name string,
+	pre_rule_uid string, post_rule_name string, post_rule_uid string) {
 
-		backupDriver := Inst().Backup
-		bkpCreateRequest := &api.BackupCreateRequest{
-			CreateMetadata: &api.CreateMetadata{
-				Name:  backupName,
-				OrgId: orgID,
-			},
-			BackupLocationRef: &api.ObjectRef{
-				Name: bLocation,
-				Uid:  bLocationUID,
-			},
-			Cluster:        clusterName,
-			Namespaces:     namespaces,
-			LabelSelectors: labelSelectors,
+	var bkp *api.BackupObject
+	var bkp_uid string
+	backupDriver := Inst().Backup
+	bkpCreateRequest := &api.BackupCreateRequest{
+		CreateMetadata: &api.CreateMetadata{
+			Name:  backupName,
+			OrgId: orgID,
+		},
+		BackupLocationRef: &api.ObjectRef{
+			Name: bLocation,
+			Uid:  bLocationUID,
+		},
+		Cluster:        clusterName,
+		Namespaces:     namespaces,
+		LabelSelectors: labelSelectors,
+		ClusterRef: &api.ObjectRef{
+			Name: clusterName,
+			Uid:  uid,
+		},
+		PreExecRuleRef: &api.ObjectRef{
+			Name: pre_rule_name,
+			Uid:  pre_rule_uid,
+		},
+		PostExecRuleRef: &api.ObjectRef{
+			Name: post_rule_name,
+			Uid:  post_rule_uid,
+		},
+	}
+	//ctx, err := backup.GetPxCentralAdminCtx()
+	ctx, err := backup.GetAdminCtxFromSecret()
+	log.FailOnError(err,  "Fetching px-central-admin ctx")
+
+	_, err = backupDriver.CreateBackup(ctx, bkpCreateRequest)
+	log.FailOnError(err, "Taking backup of applications")
+
+	bkpEnumerateReq := &api.BackupEnumerateRequest{
+		OrgId: orgID}
+	curBackups, err := backupDriver.EnumerateBackup(ctx, bkpEnumerateReq)
+	for _, bkp = range curBackups.GetBackups() {
+		if bkp.Name == backupName {
+			bkp_uid = bkp.Uid
+			break
 		}
-		//ctx, err := backup.GetPxCentralAdminCtx()
-		ctx, err := backup.GetAdminCtxFromSecret()
-		Expect(err).NotTo(HaveOccurred(),
-			fmt.Sprintf("Failed to fetch px-central-admin ctx: [%v]",
-				err))
-		_, err = backupDriver.CreateBackup(ctx, bkpCreateRequest)
-		Expect(err).NotTo(HaveOccurred(),
-			fmt.Sprintf("Failed to create backup [%s] in org [%s]. Error: [%v]",
-				backupName, orgID, err))
-	})
+	}
+	backupInspectRequest := &api.BackupInspectRequest{
+		Name:  backupName,
+		Uid:   bkp_uid,
+		OrgId: orgID,
+	}
+	time.Sleep(time.Minute * 2)
+	resp, err := backupDriver.InspectBackup(ctx, backupInspectRequest)
+	dash.VerifyFatal(resp.GetBackup().GetStatus().Status, api.BackupInfo_StatusInfo_Success, "Inspecting the backup taken")
+	log.FailOnError(err, "Inspecting the backup taken")
```

Comment:
check for the error first then verification, in this case if there is a err resp. wil give nil pointer panic

Reply:
done